engine/spec/components/targeted_content_spec.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples "public_targeted_content" do
+  it "renders" do
+    expect(component.at(".cads-targeted-content")).to be_present
+  end
+
+  it "has no label" do
+    expect(component.at(".cads-badge")).not_to be_present
+  end
+end
+
+RSpec.shared_examples "adviser_targeted_content" do
+  let(:badge) { component.at(".cads-badge") }
+
+  it "renders an adviser variant" do
+    expect(component.at(".cads-targeted-content--adviser")).to be_present
+  end
+
+  it "has adviser label" do
+    expect(badge.text).to eq "Adviser"
+  end
+
+  context "when welsh language" do
+    before { I18n.locale = :cy }
+    it "has translated label" do
+      expect(badge.text).to eq "Cynghorydd"
+    end
+  end
+end
+
+RSpec.describe CitizensAdviceComponents::TargetedContent, type: :component do
+  subject(:component) do
+    render_inline(
+      described_class.new(
+        id: "targeted-content-example",
+        type: type.presence,
+        title: title.presence,
+        heading_level: heading_level.presence
+      )
+    ) do
+      "Example content"
+    end
+  end
+
+  let(:type) { :public }
+  let(:title) { "Example title" }
+  let(:heading_level) { nil }
+
+  it "renders content block" do
+    expect(component.text).to include "Example content"
+  end
+
+  it_behaves_like "public_targeted_content"
+
+  it "has a h2 title by default" do
```

Comment:
`has a correct h2 title`

Reply:
I think I'm happy with this wording. Intending to capture that we default to a `h2` if no `heading_level` is provided.

Reply:
hang on though, that's not validating that here. If we want to validate that shouldn't it validate that the h2 is present? By doing something like

`expect(component.at("h2")).not_to be_nil`